google/resource_pubsub_subscription_iam_test.go

```diff
@@ -0,0 +1,246 @@
+			{
+				// Test IAM Binding creation
+				Config: testAccPubsubSubscriptionIamBinding_basic(subscription, topic, account),
+				Check: testAccCheckPubsubSubscriptionIam(subscription, "roles/pubsub.subscriber", []string{
+					fmt.Sprintf("serviceAccount:%s-1@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+				}),
+			},
+			{
+				// Test IAM Binding update
+				Config: testAccPubsubSubscriptionIamBinding_update(subscription, topic, account),
+				Check: testAccCheckPubsubSubscriptionIam(subscription, "roles/pubsub.subscriber", []string{
+					fmt.Sprintf("serviceAccount:%s-1@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+					fmt.Sprintf("serviceAccount:%s-2@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+				}),
+			},
+			{
+				ResourceName:  "google_pubsub_subscription_iam_binding.foo",
+				ImportStateId: fmt.Sprintf("%s roles/pubsub.subscriber", getComputedSubscriptionName(getTestProjectFromEnv(), subscription)),
+				ImportState:   true,
+			},
+		},
+	})
+}
+
+func TestAccPubsubSubscriptionIamMember(t *testing.T) {
+	t.Parallel()
+
+	topic := "test-topic-iam-" + acctest.RandString(10)
+	subscription := "test-subscription-iam-" + acctest.RandString(10)
+	account := "test-iam-" + acctest.RandString(10)
+	accountEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test Iam Member creation (no update for member, no need to test)
+				Config: testAccPubsubSubscriptionIamMember_basic(subscription, topic, account),
+				Check: testAccCheckPubsubSubscriptionIam(subscription, "roles/pubsub.subscriber", []string{
+					fmt.Sprintf("serviceAccount:%s", accountEmail),
+				}),
+			},
+			{
+				ResourceName:  "google_pubsub_subscription_iam_member.foo",
+				ImportStateId: fmt.Sprintf("%s roles/pubsub.subscriber serviceAccount:%s", getComputedSubscriptionName(getTestProjectFromEnv(), subscription), accountEmail),
+				ImportState:   true,
+			},
+		},
+	})
+}
+
+func TestAccPubsubSubscriptionIamPolicy(t *testing.T) {
+	t.Parallel()
+
+	topic := "test-topic-iam-" + acctest.RandString(10)
+	subscription := "test-subscription-iam-" + acctest.RandString(10)
+	account := "test-iam-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPubsubSubscriptionIamPolicy_basic(subscription, topic, account, "roles/pubsub.subscriber"),
+				Check: testAccCheckPubsubSubscriptionIam(subscription, "roles/pubsub.subscriber", []string{
+					fmt.Sprintf("serviceAccount:%s@%s.iam.gserviceaccount.com", account, getTestProjectFromEnv()),
+				}),
+			},
+			{
+				Config: testAccPubsubSubscriptionIamPolicy_basic(subscription, topic, account, "roles/pubsub.subscriber"),
```

Comment:
What's the difference between this step and the previous step?

Reply:
Good catch, I changed the role for the second step.